docs/protos.md

```diff
@@ -15,13 +15,38 @@ On an Ubuntu-based system, install the following dependencies.
 sudo apt install build-essential protobuf-compiler golang
 ```
 
+## Protobuf programming practices
+
+You should follow standard [protobuf programming practices] when developing
+a protobuf definition. In addition, we establish the following practices.
+
+### Package versioning
+
+To enable consumers to support multiple versions of the in-toto attestation
+spec or predicates[^1], we maintain versioned sub-packages for the protos.
+
+This means, the protos for a new major version should be placed under new
+`vMAJ` sub-package under the respective protos package:
+
+-   spec in `protos/in\_toto\_attestation/`
+-   predicates in `protos/in\_toto\_attestation/<predicate>/`
+
+### Updates
+
+Minor version updates to the protobufs are expected to be fully backwards
+compatible (per [semver guidelines]), and should be made directly to the
+proto defintions in the corresponding `vMAJ` sub-package.
+
+To ensure backwards compatibility of the updated definition, please review
+the message type [update guidelines].
+
 ## Regenerating Go proto libraries
 
 [It's typical to keep generated Go code in the repository itself](https://go.dev/doc/articles/go_command#:~:text=and%20then%20check%20those%20generated%20source%20files%20into%20your%20repository)
 since it makes users' lives much easier.
 
 Proto libraries should be regenerated & committed after any change to the
-proto files:
+proto files. From the root directory, run:
 
 ```shell
 $ make protos
```

Comment:
this has changed and needs to be updated. I also think it may be better to remove the git commit line, we probably want better commit messages describing the actual changes...

Reply:
nvm, i missed the double makefiles